salt/_modules/metalk8s_cordon.py

```diff
@@ -99,4 +99,13 @@ def node_uncordon(node_name, **kwargs):
 
         salt '*' kubernetes.node_uncordon  node_name="minikube"
     '''
-    return node_set_unschedulable(node_name, False, **kwargs)
+    # need to remove all taints before node uncordon
+    set_no_taints = []
+    remove_all_taint = __salt__['metalk8s_kubernetes.node_set_taints'](
+        node_name, set_no_taints, **kwargs
+    )
+
+    if remove_all_taint is not None:
+        return node_set_unschedulable(node_name, False, **kwargs)
```

Comment:
Not in an if you need to remove unschedulable always.
You ~may~ have some version (1.11) where you will have basically no taint to remove normally.

The module should remove the taint `node.kubernetes.io/unschedulable` (and only this one) AND set `unschedulable` flag to `False` (even if no taint get removed)